File: 프로그래머스/1/12947. 하샤드 수/하샤드 수.java
class Solution {
    public boolean solution(int x) {
        boolean answer = true;
        String s = Integer.toString(x);
        String[] str = s.split("");
        int num = 0;
        
        for(int i = 0; i < str.length; i++) {
            num += Integer.parseInt(str[i]);
        }
        if(x % num  != 0) {
            answer = false;
        } 
        return answer;
    }
}

File: 프로그래머스/1/12947. 하샤드 수/하샤드 수.js
function solution(x) {
    var answer = true;
    
    var str = x.toString();
    var sum = parseInt(str[0]);
    
    for(let i=1 ; i< str.length ;i++){
      sum += parseInt(str[i]);
    }

    result = x % sum ? false : true
    return result;
}